CMakeLists.txt

```diff
@@ -1,11 +1,12 @@
 cmake_minimum_required (VERSION 3.18)
 
-set(CMAKE_CXX_STANDARD 14)
+set(CMAKE_CXX_STANDARD 17)
```

Comment:
@tepperly To minimize disruptions on our users, I would like to keep this at 14. Would cmake-ing HiOp with `-DCMAKE_CXX_STANDARD=17` work on your end?